### PR TITLE
feat: paper configs

### PR DESCRIPTION
**Pull request by clement-bonnet**
- up `BinPack` version to v2 with configs to make it faster
- up `Connector` version to v2 with the `RandomWalkGenerator` that guarantees an optimal `prop_connections` of 1
- add `step_count` to the observation of `MMST`
- fix `Tetris` doc
- fix `Knapsack` configs (50 items instead of 20) -> I do not up the version since there has been no releases that includes this bug
- set `time_limit` of `Tetris` to 400 instead of 4000
- fix `Tetris` termination and termination reward
- improve `MMST` speed by parallelizing some operations
- add all environments to training configs
- update all training configs to match the paper
- add `step_count` to `MMST` network
- improve `Tetris` network

### CODE REVIEW

File: jumanji/environments/routing/mmst/env.py [hunk 1]
```diff
@@ -574,14 +578,19 @@ def done_fun(
...
-            )
-            finished_agents = finished_agents.at[agent].set(finished)
+        # finished_agents = jnp.zeros_like(state.finished_agents)
+        # for agent in range(self.num_agents):
+        #     finished = done_fun(
+        #         state.nodes_to_connect[agent],
+        #         state.connected_nodes[agent],
+        #         self.num_nodes_per_agent,
+        #     )
+        #     finished_agents = finished_agents.at[agent].set(finished)
```
Comment (clement-bonnet) [inline jumanji/environments/routing/mmst/env.py:588]:
```suggestion
```

File: jumanji/environments/routing/mmst/env.py [hunk 2]
```diff
@@ -596,14 +597,19 @@ def num_connections(
...
-            )
-            connections = connections.at[agent].set(total_ratio)
+        # connections = jnp.zeros((self.num_agents, 2))
+        # for agent in range(self.num_agents):
+        #     total_ratio = num_connections(
+        #         state.nodes_to_connect[agent],
+        #         state.connected_nodes[agent],
+        #         self.num_nodes_per_agent,
+        #     )
+        #     connections = connections.at[agent].set(total_ratio)
```
Comment (clement-bonnet) [inline jumanji/environments/routing/mmst/env.py:607]:
```suggestion
```

**Review by ulricharmel — DISMISSED** [review]
The mmst changes looks good to me.

File: README.md [hunk 3]
```diff
@@ -80,20 +80,20 @@ problems.
...
 |------------------------------------------|----------|------------------------------------------------------|--------------------------------------------------------------------------------------------------|------------------------------------------------------------------------|
 | 🔢 Game2048                              | Logic  | `Game2048-v1`                                        | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/game_2048/)   | [doc](https://instadeepai.github.io/jumanji/environments/game_2048/)   |
-| 🔵🔗🟡🔗🔴 GraphColoring                              | Logic  | `GraphColoring-v0`                                        | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/graph_coloring/)   | [doc](https://instadeepai.github.io/jumanji/environments/graph_coloring/)   |
+| 🔵🔗🟡🔗🔴 GraphColoring                              | Logic  | `GraphColoring-v0`                                   | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/graph_coloring/)   | [doc](https://instadeepai.github.io/jumanji/environments/graph_coloring/)   |
```
Comment (dluo96) [inline README.md:83]: If possible, could we stick with just one emoji? On the README, it looks a bit clunky to have 5 emojis for GraphColoring but one for all other environments

File: README.md [hunk 4]
```diff
@@ -80,20 +80,20 @@ problems.
...
-| ✏️ Sudoku                       | Logic    | `Sudoku-v0` <br/>`Sudoku-very-easy-v0`| [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/sudoku/) | [doc](https://instadeepai.github.io/jumanji/environments/sudoku/) |
-| 📦 BinPack (3D BinPacking Problem)       | Packing  | `BinPack-v1`                                         | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/packing/bin_pack/)  | [doc](https://instadeepai.github.io/jumanji/environments/bin_pack/)    |
+| ✏️ Sudoku                       | Logic    | `Sudoku-v0` <br/>`Sudoku-very-easy-v0`               | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/sudoku/) | [doc](https://instadeepai.github.io/jumanji/environments/sudoku/) |
+| 📦 BinPack (3D BinPacking Problem)       | Packing  | `BinPack-v2`                                         | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/packing/bin_pack/)  | [doc](https://instadeepai.github.io/jumanji/environments/bin_pack/)    |
```
Comment (dluo96) [inline README.md:87]: We are upping the version of BinPack because we've changed the default configs, right?

File: jumanji/environments/packing/knapsack/env.py [hunk 5]
```diff
@@ -103,7 +103,7 @@ def __init__(
         """
 
         self.generator = generator or RandomGenerator(
-            num_items=20,
+            num_items=50,
```
Comment (dluo96) [inline jumanji/environments/packing/knapsack/env.py:106]: If we are changing num_items for Knapsack, should we also increment the version in the registry?

File: jumanji/environments/packing/tetris/env.py [hunk 6]
```diff
@@ -203,7 +201,8 @@ def step(
         # In case the grid is empty the color should be set 0.
         color = jnp.array([1, grid_padded.max()])
         colored_tetromino = tetromino * jnp.max(color)
-        reward = self.reward_list[nbr_full_lines]
+        is_valid = state.action_mask[tuple(action)]
+        reward = self.reward_list[nbr_full_lines] * is_valid
```
Comment (dluo96) [inline jumanji/environments/packing/tetris/env.py:205]: 0 reward if the action is invalid?

File: jumanji/environments/routing/mmst/env.py [hunk 7]
```diff
@@ -388,43 +400,35 @@ def _state_to_timestep(
...
-        def make_transition_timestep(state: State) -> TimeStep:
+        def make_transition_timestep() -> TimeStep[Observation]:
             return transition(
                 reward=reward,
                 observation=observation,
                 extras=extras,
             )
 
-        is_done = state.finished_agents.all()
+        agents_are_done = state.finished_agents.all()
```
Comment (dluo96) [inline jumanji/environments/routing/mmst/env.py:425]: Why agents in plural if there is only a single agent (indicated by previous line of code)?

File: jumanji/environments/routing/mmst/viewer_test.py [hunk 8]
```diff
@@ -42,23 +39,9 @@ def test__animation(deterministic_mmst_env: Tuple[MMST, State, TimeStep]) -> Non
...
-        jnp.array([6, 8]),
-        jnp.array([10, 10]),
-        jnp.array([9, 4]),
-    ]
-
-    for action in actions:
-        state, _ = step_fn(state, action)
-        states.append(state)
-
+    states = 2 * [state]
```
Comment (dluo96) [inline jumanji/environments/routing/mmst/viewer_test.py:44]: Why `2 *`?

File: jumanji/training/configs/env/game_2048.yaml [hunk 9]
```diff
@@ -1,5 +1,5 @@
 name: game_2048
-registered_version: Game2048-v0
+registered_version: Game2048-v1
```
Comment (dluo96) [inline jumanji/training/configs/env/game_2048.yaml:2]: Remind me why we are incrementing this - was it because of the bug that was found?

File: README.md (hunk 3 above)
Comment (clement-bonnet) [reply README.md:83]: Agree
```suggestion
| 🔵 GraphColoring                              | Logic  | `GraphColoring-v0`                                   | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/graph_coloring/)   | [doc](https://instadeepai.github.io/jumanji/environments/graph_coloring/)   |
```

File: README.md (hunk 4 above)
Comment (clement-bonnet) [reply README.md:87]: Yes

File: jumanji/environments/packing/knapsack/env.py (hunk 5 above)
Comment (clement-bonnet) [reply jumanji/environments/packing/knapsack/env.py:106]: This point is addressed in the PR description. Let me know if this works for you.

File: jumanji/environments/routing/mmst/env.py (hunk 7 above)
Comment (clement-bonnet) [reply jumanji/environments/routing/mmst/env.py:425]: The state attribute is called `finished_agents`, so I want to improve consistency by calling the `done` an `agents_are_done` variable. Does it make sense?

File: jumanji/environments/routing/mmst/viewer_test.py (hunk 8 above)
Comment (ulricharmel) [reply jumanji/environments/routing/mmst/viewer_test.py:44]: This is testing the animation. So we just want to pass a list of states and see if the animation code runs. It doesn't really matter if the states are different or not.

Comment (clement-bonnet) [reply jumanji/environments/routing/mmst/viewer_test.py:44]: The test was taking ages and does not require all previously generated states to be rendered, I believe only 2 states are sufficient to test `animate`.

File: jumanji/training/configs/env/game_2048.yaml (hunk 9 above)
Comment (clement-bonnet) [reply jumanji/training/configs/env/game_2048.yaml:2]: It was already incremented, this fixes the training configs. It was incremented due to a bug in the action mask in #144

File: jumanji/training/configs/env/connector.yaml [hunk 10]
```diff
@@ -9,18 +9,18 @@ network:
...
 evaluation:
-    eval_total_batch_size: 5000
-    greedy_eval_total_batch_size: 5000
+    eval_total_batch_size: 10000
+    greedy_eval_total_batch_size: 10000
 
 a2c:
     normalize_advantage: False
-    discount_factor: 0.99
+    discount_factor: 1.0
```
Comment (TristanKalloniatis) [inline jumanji/training/configs/env/connector.yaml:23]: No discounting?

File: jumanji/training/configs/env/tsp.yaml [hunk 11]
```diff
@@ -24,5 +24,5 @@ a2c:
     bootstrapping_factor: 0.95
     l_pg: 1.0
     l_td: 1.0
-    l_en: 0.01
+    l_en: 0.005
```
Comment (TristanKalloniatis) [inline jumanji/training/configs/env/tsp.yaml:27]: Btw you are giving some numbers like this as decimals and others in scientific notation, any reason?

File: README.md [hunk 12]
```diff
@@ -80,7 +80,7 @@ problems.
...
 |------------------------------------------|----------|------------------------------------------------------|--------------------------------------------------------------------------------------------------|------------------------------------------------------------------------|
 | 🔢 Game2048                              | Logic  | `Game2048-v1`                                        | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/game_2048/)   | [doc](https://instadeepai.github.io/jumanji/environments/game_2048/)   |
-| 🔵🔗🟡🔗🔴 GraphColoring                              | Logic  | `GraphColoring-v0`                                   | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/graph_coloring/)   | [doc](https://instadeepai.github.io/jumanji/environments/graph_coloring/)   |
+| 🔵 GraphColoring                              | Logic  | `GraphColoring-v0`                                   | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/graph_coloring/)   | [doc](https://instadeepai.github.io/jumanji/environments/graph_coloring/)   |
```
Comment (dluo96) [inline README.md:83]: How about this one?
```suggestion
| 🎨 GraphColoring                              | Logic  | `GraphColoring-v0`                                   | [code](https://github.com/instadeepai/jumanji/tree/main/jumanji/environments/logic/graph_coloring/)   | [doc](https://instadeepai.github.io/jumanji/environments/graph_coloring/)   |
```

File: jumanji/environments/packing/knapsack/env.py (hunk 5 above)
Comment (dluo96) [reply jumanji/environments/packing/knapsack/env.py:106]: Ah yes, gotcha.

File: jumanji/environments/packing/tetris/env.py (hunk 6 above)
Comment (dluo96) [reply jumanji/environments/packing/tetris/env.py:205]: Ok 👍

File: jumanji/environments/routing/mmst/env.py (hunk 7 above)
Comment (dluo96) [reply jumanji/environments/routing/mmst/env.py:425]: Sounds good 👍

File: jumanji/environments/routing/mmst/viewer_test.py (hunk 8 above)
Comment (dluo96) [reply jumanji/environments/routing/mmst/viewer_test.py:44]: Understood 👍

File: jumanji/training/configs/env/game_2048.yaml (hunk 9 above)
Comment (dluo96) [reply jumanji/training/configs/env/game_2048.yaml:2]: Ok 👍

**Review by dluo96 — DISMISSED** [review]
LGTM

File: jumanji/environments/routing/mmst/env.py [hunk 13]
```diff
@@ -596,14 +597,11 @@ def num_connections(
...
 
-        connections = jnp.zeros((self.num_agents, 2))
-        for agent in range(self.num_agents):
-            total_ratio = num_connections(
-                state.nodes_to_connect[agent],
-                state.connected_nodes[agent],
-                self.num_nodes_per_agent,
-            )
-            connections = connections.at[agent].set(total_ratio)
+        connections = jax.vmap(num_connections, in_axes=(0, 0, None))(
```
Comment (djbyrne) [inline jumanji/environments/routing/mmst/env.py:600]: maybe outside the scope of this MR, but I would rename "num_connections" to something like get/calc _num_connections. Or even throw `_func` at the end of it

File: jumanji/training/configs/env/bin_pack.yaml [hunk 14]
```diff
@@ -8,20 +8,20 @@ network:
...
+    greedy_eval_total_batch_size: 10000
 
 a2c:
     normalize_advantage: False
     discount_factor: 1.0
     bootstrapping_factor: 0.95
     l_pg: 1.0
     l_td: 1.0
-    l_en: 0.01
+    l_en: 0.005
```
Comment (djbyrne) [inline jumanji/training/configs/env/bin_pack.yaml:26]: Nit: would rename the loss coefficients to be more descriptive, but outside the scope of this MR

File: jumanji/training/configs/env/tetris.yaml [hunk 15]
```diff
@@ -2,15 +2,15 @@ name: tetris
 registered_version: Tetris-v0
 
 network:
-    conv_num_channels: 32
-    tetromino_layers: 16
-    final_layer_dims: [128,128]
+    conv_num_channels: 64
+    tetromino_layers: [16, 16]
```
Comment (djbyrne) [inline jumanji/training/configs/env/tetris.yaml:6]: what does "tetromino" mean?

File: jumanji/training/configs/env/tsp.yaml (hunk 11 above)
Comment (clement-bonnet) [reply jumanji/training/configs/env/tsp.yaml:27]: No reason, I don't have a strong opinion at all. Please feel free to suggest changes where you think it should be different. Otherwise, I will go with what is in if you don't mind.

File: jumanji/environments/routing/mmst/env.py (hunk 13 above)
Comment (clement-bonnet) [reply jumanji/environments/routing/mmst/env.py:600]: That's a good point. I agree! For priority reasons, I will keep this way in this PR.

File: jumanji/training/configs/env/bin_pack.yaml (hunk 14 above)
Comment (clement-bonnet) [reply jumanji/training/configs/env/bin_pack.yaml:26]: At least we should have a readme :laughing: describing these arguments. I will make an issue for that.

File: jumanji/training/configs/env/tetris.yaml (hunk 15 above)
Comment (clement-bonnet) [reply jumanji/training/configs/env/tetris.yaml:6]: Please see the Tetris wiki [here](https://tetris.fandom.com/wiki/Tetromino)

**Review by dluo96 — APPROVED** [review]

File: jumanji/training/configs/env/tetris.yaml (hunk 15 above)
Comment (djbyrne) [reply jumanji/training/configs/env/tetris.yaml:6]: Well I feel silly...
